source/Calamari.Common/Plumbing/Pipeline/IBehaviourExtensions.cs

```diff
@@ -12,8 +12,19 @@ public static Task CompletedTask(this IBehaviour _)
 #elif NET452
             return Task.FromResult(0);
 #else
-            return new Task<int>(() => 0);;
+            return Net40CompletedTask;
 #endif
         }
+
+#if NET40
+        static readonly Task Net40CompletedTask = CreateNet40CompletedTask();
+
+        static Task<int> CreateNet40CompletedTask()
+        {
+            var taskCompletionSource = new TaskCompletionSource<int>();
```

Comment:
This is the fix: using a `TaskCompletionSource` to create a completed task in net40